website/docs/authenticating_via_msi.html.markdown

```diff
@@ -0,0 +1,97 @@
+---
+layout: "azurerm"
+page_title: "AzureRM: Authenticating via Managed Service Identity"
+sidebar_current: "docs-azurerm-index-authentication-msi"
+description: |-
+  The Azure Resource Manager provider supports authenticating via multiple means. This guide will cover configuring a Managed Service Identity which can be used to access Azure Resource Manager.
+
+---
+
+# Authenticating to Azure Resource Manager using Managed Service Identity
+
+Terraform supports authenticating to Azure through Managed Service Identity, Service Principal or the Azure CLI.
+
+We recommend using Managed Service Identity when running in a Shared Environment (such as within a CI server/automation) that you do not wish to configure credentials for - and [authenticating via the Azure CLI](authenticating_via_azure_cli.html) when you're running Terraform locally. Note that managed service identity is only available for virtual machines within Azure.
+
+## Configuring Managed Service Identity
+
+Managed Service Identity allows an Azure virtual machine to retrieve a token to access the Azure API without needing to pass in credentials. This works by creating a service principal in Azure Active Directory that is associated to a virtual machine. This service principal can then be granted permissions to Azure resources.
+There are various ways to configure managed service identity - see the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/active-directory/msi-overview) for details.
+You can then run Terraform from the MSI enabled virtual machine by setting the use_msi provider option to true.
+
+### Configuring Managed Service Identity using Terraform
+
+Managed service identity can also be configured using Terraform. The following template shows how. Note that for a Linux VM you must use the ManagedIdentityExtensionForLinux.
+
+```hcl
+resource "azurerm_virtual_machine" "virtual_machine" {
```

Comment:
Can you put a Linux example ? This example creates WindowsServer and the documentation above says about Linux.

Reply:
As noted, for Linux it would be exactly the same except you use ManagedIdentityExtensionForLinux instead of ManagedIdentityExtensionForWindows extension